### ISSUE TITLE
Proposal: Change `var` and `const` for better readability

### ISSUE THREAD

**Issue by user00e01** [ziglang/zig#5396]
# Proposal: Change `var` and `const` for better readability

## Introduction

***The main benefit of this proposal is making programming in Zig more accessible***

Take the following code example:
```zig
const my_constant = 420;
var my_variable = 69;
```
Assume that the person reading this code has never programmed before. Later, said person might come back to this code and wonder what the difference is between `const` and `var`.

[Based on a quadruple blind study performed by Oxford in 2012](https://www.youtube.com/watch?v=dQw4w9WgXcQ), it has been proven that many people who do not program cannot distinguish the unique features of constants and variables as they do not understand them fully.

I propose that `var` and `const` be replaced with `fluid` and `solid`, respectively. Now, when someone reads the code, no matter their background or experience, they'll be able to distinguish that variables are "fluid", or can be changed, and that constants are "solid", and cannot.

Now, our example code looks like this:
```zig
solid my_constant = 420;
fluid my_variable = 69;
```

Also, I like how the `solid` and `fluid` terms align, it just looks better.

## Problems

There are a few problems with this approach. Firstly, `solid` and `fluid` are long words to type, so they might have to be replaced with `s` and `f` accordingly. Secondly, people might want to use `liquid` instead of `fluid`, so that'll have to be discussed further in this issue.

These are the only problems I see with this approach.

## Implementation

This could be implemented with simple compiler changes. I believe that `var` and `const` shouldn't be immediately phased out, but progressively removed in order to maintain compatibility with older, less inclusive codebases.

## Prior Art

[This language you've never heard of that I and 3 other people use](https://www.youtube.com/watch?v=6n3pFFPSlW4) has implemented `solid` and `fluid` variables before, so that's why Zig should do the same.

It has also implemented similar, related features such as solid melting, and liquid evaporation, which might be considered in another issue.

## Conclusion

In conclusion, replacing `var` and `const` with `fluid` and `solid` would make Zig more accessible and welcoming to beginners.

## Hypothetical Uses

* Lays the foundation for matter-oriented programming, which could be an interesting feature of Zig
* Evaporation of liquids
* Melting of solids

**Comment by user00e01** [comment]
Also, if you didn't notice, this issue is not real.

**Comment by user00e00** [comment]
1: Impersonation is a shitty thing to do, even when it's done well and arguably in a funny way.
2: I'll take this as a hint that my proposals are considered useless by people in the zig community. Fair enough. I'll stop wasting your time and my time as well.

**Comment by andrewrk** [comment]
@user00e00 the impersonating account has been blocked and this issue will be deleted because it is spam. Sorry that happened to you. Please don't consider this person to be representative of the zig community. This behavior is not welcome here.
